### ISSUE TITLE
code coverage on travis

### ISSUE THREAD

**Issue by prasadtalasila** [prasadtalasila/BITS-Darshini#139]
The [coveralls plugin](https://github.com/trautonen/coveralls-maven-plugin) can help in pushing the code coverage report to coveralls. We can try to use the same.

There is also [codacy plugin](https://github.com/codacy/codacy-coverage-reporter) if coveralls plug in does not work.

**Comment by prasadtalasila** [comment]
update: our pom.xml file already has the necessary configuration for coveralls plugin with jacoco code coverage tool. We just need to follow [the suggested configuration](https://github.com/trautonen/coveralls-maven-plugin#jacoco) to see if the code coverage is reported to code climate.

### Alternative ###
The new [code climate test reporter](https://docs.codeclimate.com/v1.0/docs/configuring-test-coverage) has support for Java. Please see the example configuration for [tavis-ci](https://docs.codeclimate.com/v1.0/docs/travis-ci-test-coverage)

**Comment by prasadtalasila** [comment]
The codeclimate test reporter ID token has been placed in travis CI settings of the project. The token can be accessed using the environment variable **$CC_TEST_REPORTER_ID**.

**Comment by prasadtalasila** [comment]
We found [codecov](codecov.com) to be a better code coverage reporter. Let's integrate that plug-in for code coverage.

**Comment by prasadtalasila** [comment]
@kapilkalra04 I merged the existing PR #147. There is still some old code in pom.xml. See the following code snippet.

```
				<plugin>
					<groupId>org.jacoco</groupId>
					<artifactId>jacoco-maven-plugin</artifactId>
					<version>0.7.9</version>
					<executions>
						<execution>
							<id>prepare-agent</id>
							<goals>
								<goal>prepare-agent</goal>
							</goals>
						</execution>
					</executions>
				</plugin>
```

This part can be safely removed now. Please fork my dev branch again, and update the pom.xml and send a PR again.

**Comment by prasadtalasila** [comment]
PR #147 solves this issue.
